launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/corebuild/CoreBuildMainTab.java

```diff
@@ -10,73 +10,51 @@
  *******************************************************************************/
 package org.eclipse.cdt.launch.ui.corebuild;
 
-import org.eclipse.cdt.launch.internal.ui.LaunchUIPlugin;
-import org.eclipse.core.resources.IProject;
-import org.eclipse.core.resources.IResource;
-import org.eclipse.core.runtime.CoreException;
+import org.eclipse.cdt.launch.internal.ui.LaunchMessages;
+import org.eclipse.cdt.launch.ui.CMainTab2;
 import org.eclipse.debug.core.ILaunchConfiguration;
-import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
-import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
-import org.eclipse.swt.SWT;
-import org.eclipse.swt.layout.GridData;
-import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Label;
-import org.eclipse.swt.widgets.Text;
 
 /**
  * @since 9.1
  */
-public class CoreBuildMainTab extends AbstractLaunchConfigurationTab {
-
-	private Text projectName;
-
-	@Override
-	public void createControl(Composite parent) {
-		Composite comp = new Composite(parent, SWT.NONE);
-		comp.setLayout(new GridLayout());
-
-		Label label = new Label(comp, SWT.NONE);
-		label.setText("This launch configuration was automatically created.");
-		label.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-
-		label = new Label(comp, SWT.NONE);
-		label.setText("Project:");
-
-		projectName = new Text(comp, SWT.READ_ONLY | SWT.BORDER);
-		projectName.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-
-		setControl(comp);
-	}
+public class CoreBuildMainTab extends CMainTab2 {
```

Comment:
O: I'm not sure about this change.
CMainTab2 says "This class is not intended to be subclassed."

Reply:
That line was written in 2008, before Core Build was active. I thought it would be best to reuse the class. Otherwise I would have to duplicate a lot of code.